### PR TITLE
Word alternatives

### PR DESCRIPTION
**Pull request by Focuson-github**
### Summary

In order to get "word_alternatives" from the results in the module of "speech to text", add code "private List wordAlternatives;" in Transcript.java. then add the SpeechWordAlternative.java and WordAlternative.java for accept json String of result. Also, the "WordAlternativeTypeAdapter.java" is for parsing json object

### Other Information

Something wrong with merge request "Word alternatives #390", so i closed the merger request. This is the new merge request.

### CODE REVIEW

File: src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/WordAlternative.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.ibm.watson.developer_cloud.speech_to_text.v1.model;
+
+import com.google.gson.annotations.JsonAdapter;
+import com.ibm.watson.developer_cloud.service.model.GenericModel;
+
+/**
+ * Created by Focuson on 16-7-13.
+ */
+@JsonAdapter(WordAlternativeTypeAdapter.class)
```
Comment (germanattanasio) [inline src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/WordAlternative.java]: We don't need a `TypeAdapter`

File: src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/WordAlternative.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.ibm.watson.developer_cloud.speech_to_text.v1.model;
+
+import com.google.gson.annotations.JsonAdapter;
+import com.ibm.watson.developer_cloud.service.model.GenericModel;
+
+/**
+ * Created by Focuson on 16-7-13.
```
Comment (germanattanasio) [inline src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/WordAlternative.java]: Don't add owner or date to the Javadoc. Describe what the class represents.

File: src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeechWordAlternative.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.google.gson.annotations.SerializedName;
+import com.ibm.watson.developer_cloud.service.model.GenericModel;
+
+import java.util.List;
+
+/**
+ * Created by Focuson on 16-7-13.
+ */
+public class SpeechWordAlternative extends GenericModel {
```
Comment (germanattanasio) [inline src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeechWordAlternative.java]: I think `WordAlternative` is a better class name

**Comment by germanattanasio** [comment]
Hi @Focuson-github thanks for the PR. Can you also update the integration tests with some checks to make sure the code you wrote is working? You should test that the word alternatives are there.

**Comment by germanattanasio** [comment]
I'm going to merge this but modify it to address my comments above

File: src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeechWordAlternative.java (hunk 3 above)
Comment (Focuson-github) [inline src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/SpeechWordAlternative.java]: The `WordAlternative` is already exits

**Comment by Focuson-github** [comment]
@germanattanasio  the test result as follow:
`[{
  "final": true,
  "alternatives": [
    {
      "confidence": 0.304,
      "transcript": "那个 中国 丝绸 网 的 商人 进出 的 吗 "
    }
  ],
  "word_alternatives": []
}, {
  "final": true,
  "alternatives": [
    {
      "confidence": 0.501,
      "transcript": "然后 你 这个 这块 产品 开发 的 因为 投资 实习 就是 目前 投资 其他的 之后 的 反正 我们 系统 会 制度 家里 投资 的 车 还有 一起 收益 会 迟到 的 帐户 询问 当中 的 积极的 换 的 你 可以 关注 一下 你 丈夫 帐目 情况 他 投资 时间 的话 你就 十五 天 "
    }
  ],
  "word_alternatives": [
    {
      "start_time": 16.9,
      "alternatives": [
        {
          "confidence": 0.9508,a
          "word": "一下"
        }
      ],
      "end_time": 17.14
    },
    {
      "start_time": 17.14,
      "alternatives": [
        {
          "confidence": 0.9283,
          "word": "你"
        }
      ],
      "end_time": 17.28
    }
  ]
}, {
  "final": true,
  "alternatives": [
    {
      "confidence": 0.551,
      "transcript": "然后 听到 你 丈夫 当中 之后 特别 的话 的 可以用 一头 其他 他的 火车 缓缓 的 体现 体力 也 都可以 操作 "
    }
  ],
  "word_alternatives": []
}, {
  "final": true,
  "alternatives": [
    {
      "confidence": 0.491,
      "transcript": "哦 好 较好 这对 因为 的话 呢 就是说 西天 汇 回 退 的 退党 比较 等 配套 你 相互 如果 当中 的 饭 的 你自己 他说 就可以了 "
    }
  ],
  "word_alternatives": []
}]`

i will push the latest code in a day or two

File: src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/WordAlternative.java (hunk 1 above)
Comment (Focuson-github) [inline src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/model/WordAlternative.java]: I have remove the code in latest code

**Comment by Focuson-github** [comment]
I have put a new merge request `#408` , please help review again, thanks a lot

**Pull request merged**